context/AuthContextProvider.tsx

```diff
@@ -0,0 +1,107 @@
+'use client';
+import { createContext, useContext, useEffect, useMemo, useState } from 'react';
+import { account } from '@/api/config';
+import { useRouter } from 'next/navigation';
+
+type UserCredentials = {
+  email: string;
+  password: string;
+};
+
+type AuthContextType = {
+  isSignedIn: boolean;
+  setIsSignedIn: (isSignedIn: boolean) => void;
+  loginAccount: (user: UserCredentials) => Promise<void | Error>;
+  logoutAccount: () => Promise<void>;
+};
+
+export const AuthContext = createContext<AuthContextType | null>(null);
+
+export const AuthContextProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [isSignedIn, setIsSignedIn] = useState<boolean>(false);
+  const router = useRouter();
+
+  // Check for a current session on component mount
+  useEffect(() => {
+    const checkSession = async () => {
+      if (!isSessionInLocalStorage()) {
+        console.log('No session data available');
+        return;
+      }
+
+      try {
+        await account.getSession('current');
+        console.log('Active session found');
+        setIsSignedIn(true);
+      } catch (error) {
+        console.error('Session validation error:', error);
+        setIsSignedIn(false);
+      }
+    };
+    checkSession();
+  }, []);
+
+  // Authenticate and set session state
+  const loginAccount = async (user: UserCredentials): Promise<void | Error> => {
+    try {
+      await account.createEmailPasswordSession(user.email, user.password);
+      setIsSignedIn(true);
+    } catch (error) {
+      console.error('Login error:', error);
+      return error as Error;
+    }
+  };
+
+  // Log out and clear session state
+  const logoutAccount = async (): Promise<void> => {
+    try {
+      await account.deleteSession('current');
+      setIsSignedIn(false);
+      await router.push('/login');
+    } catch (error) {
+      console.error('Logout error:', error);
+    }
+  };
+
+  // Helper function to validate session data in local storage
+  const isSessionInLocalStorage = (): boolean => {
```

Comment:
nit: maybe consider placing this in a line above AuthContextProvider? I was honestly confused where you were grabbing isSessionInLocalStorage from here.  Also I'm curious what your intention was by making this a function.  It's only being used once in this code, could this not be inside of AuthContextProvider's useEffect itself?

Reply:
I like to keep my functions being used within the context provider inside the context provider rather than outside.

My intent in making it a function was to be used as a function .... I think it makes it very clear as to what is being done.

Reply:
That's fair, thanks for clarifying that!